src/nni_manager/training_service/common/util.ts

```diff
@@ -6,41 +6,73 @@
 import * as cpp from 'child-process-promise';
 import * as cp from 'child_process';
 import * as fs from 'fs';
-import * as os from 'os';
+import ignore from 'ignore';
 import * as path from 'path';
+import * as tar from 'tar';
 import { String } from 'typescript-string-operations';
-import { countFilesRecursively, getNewLine, validateFileNameRecursively } from '../../common/utils';
+import { validateFileName } from '../../common/utils';
 import { GPU_INFO_COLLECTOR_FORMAT_WINDOWS } from './gpuData';
 
+/**
+ * List all files in directory except those ignored by .nniignore.
+ * @param source
+ * @param destination
+ */
+export function* listDirWithIgnoredFiles(root: string, relDir: string, ignoreFiles: string[]): Iterable<string> {
+    let ignoreFile = undefined;
+    const source = path.join(root, relDir);
+    if (fs.existsSync(path.join(source, '.nniignore'))) {
```

Comment:
will check .nniignore file in every subfolder?

Reply:
Yes. Works like .gitignore.